user/hubers_functions.hpp

```diff
@@ -343,6 +445,99 @@ namespace antok {
 						std::vector<double>* _resultAddr;
 				};
 
+
+
+				//***********************************
+				// Scale Energy of a cluster depending
+				// on energy and position
+				//***********************************
+				class ExtrapNeutral : public Function
+				{
+					public:
+					ExtrapNeutral(double* XAddr, double* YAddr, double* ZAddr,
+												TLorentzVector* piLV, TLorentzVector* beamLV,
+												double* resultAddrX, double* resultAddrY)
+									:_XAddr(XAddr),_YAddr(YAddr),_ZAddr(ZAddr),
+									 _piLV(piLV), _beamLV(beamLV),
+									 _resultAddrX(resultAddrX), _resultAddrY(resultAddrY) {}
+
+					virtual ~ExtrapNeutral() {}
+
+					bool operator() () {
+						TLorentzVector gammaLV = (*_beamLV - *_piLV);
+						if(sqr(gammaLV.Z()) < 1e-13) {
+							*_resultAddrX = 9e9;
+							*_resultAddrY = 9e9;
+							return 1;
+						}
+						*_resultAddrX = gammaLV.X() / gammaLV.Z() * (*_ZAddr) + (*_XAddr);
+						*_resultAddrY = gammaLV.Y() / gammaLV.Z() * (*_ZAddr) + (*_YAddr);
+						return 1;
+					}
+
+					private:
+						double *_XAddr;
+						double *_YAddr;
+						double *_ZAddr;
```

Comment:
just `Z` is an unlucky choice for the name, something like `Zdiff` / `Zdist` would be better. This should also be changed in the generator.